lib/client.js

```diff
@@ -25,83 +51,69 @@ class Client {
     if (this.options.log_level) log.setLevel(this.options.log_level)
   }
 
-  get (path, opts) {
-    opts = opts || {}
-    opts = Object.assign({
-      cache: true,
-      authenticate: true,
-      token_expiration_retries: 1 // Number of refresh attempts to make if token expired (1 should do it!)
-    }, opts)
+  /**
+   * GET an api path
+   *
+   * @param {string} path - The path to fetch (e.g. 'current-schema/Item')
+   * @param {RequestOptions} options - A hash of options.
+   *
+   * @return {Promise} A promise that resolves the fetched data
+   */
+  get (path, options) {
+    options = Object.assign({
+      cache: true
+    }, options)
 
-    var uri = this._getFullUrl(path)
-    var cacheKey = `GET ${uri}`
+    return this._doHttpMethod('GET', path, options)
+  }
 
-    var cached = null
-    if (opts.cache && (cached = this.cache.get(cacheKey))) {
-      return Promise.resolve(cached)
-    } else {
-      var handleCache = (data) => {
-        if (opts.cache) this.cache.set(cacheKey, data)
-        return data
-      }
+  /**
+   * POST an object to an api endpoint
+   *
+   * @param {string} path - The path to fetch (e.g. 'current-schema/Item')
+   * @param {Object} body - The object/string to pass with the request
```

Comment:
Hmm... can you pass a `string` to this method? Especially since you're `stringify`ing it?

Reply:
Yeah I'm only calling `JSON.stringify` when logging (and in one other error scenario). The `body` is passed without modification directly to the `request` module, so it can be whatever you want. There's a test that confirms plaintext `body`s are posted correctly: https://github.com/NYPL-discovery/node-nypl-data-api-client/blob/3d76b83c2dbe2f116673fb550f52cc3534a83ccc/test/post-test.js#L34-L40